proto_parser/demo/demo_generator.cc

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
Split the PR into 2.

One for fixing the BUILD rule, and another one is adding the demo.